modular_skyrat/modules/modular_weapons/code/energy.dm

```diff
@@ -72,6 +72,83 @@
 /obj/item/gun/energy/laser/cfa_paladin/give_gun_safeties()
 	return
 
+
+
+// Energy SMG NX-867 Starfall (sprite changes pending) //
+
+/obj/item/gun/ballistic/automatic/laser_smg
+	name = "NTX-867 Starfall"
+	desc = "A significant improvement over older designs, this SMG is powered by cells that utilise chemical processes to generate electriciy, rather than storing it. This prototype was engineered by Sol Armistice."
+	icon_state = "oldrifle"
+	w_class = WEIGHT_CLASS_BULKY
+	inhand_icon_state = "arg"
+	mag_type = /obj/item/ammo_box/magazine/recharge/laser_smg
+	empty_indicator = TRUE
+	fire_delay = 2.25
+	can_suppress = FALSE
+	burst_size = 2
+	actions_types = list()
+	fire_sound = 'sound/weapons/laser.ogg'
+	casing_ejector = FALSE
+	bolt_type = BOLT_TYPE_OPEN
+	show_bolt_icon = FALSE
+
+/obj/item/gun/ballistic/automatic/laser_smg/Initialize(mapload)
+	. = ..()
+	bolt_locked = FALSE
+
+/obj/item/gun/ballistic/automatic/laser_smg/insert_magazine(mob/user, obj/item/ammo_box/magazine/AM, display_message)
+	. = ..()
+	
+	if(!chambered)
+		process_chamber(TRUE, FALSE, TRUE)
+
+//Lethal munitions//
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal
+	name = "energy battery NTX-867-1"
+	desc = "A reliable and rechargable battery that serves as an energy source for NTX-867 Starfall."
+	icon_state = "laser_smg_L-20"
+	base_icon_state = "laser_smg_L"
+	ammo_type = /obj/item/ammo_casing/caseless/laser_smg/lethal
+	caliber = CALIBER_LASER
+	max_ammo = 20
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal/update_desc()
+	. = ..()
+	desc = "[initial(desc)] It has [stored_ammo.len] shot\s left."
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state]-[round(ammo_count(), 4)]"
+
+/obj/item/ammo_box/magazine/recharge/laser_smg/lethal/attack_self() //No popping out the "bullets"
+	return
```

Comment:
When doing this does the code runtime at all? I know for some of the attack procs, you need to return with a certain define, but I am unsure if this is one of them.